Fix broken `TestDeltaLakeDatabricksDelete.testDeletionVectors` suite-delta-lake-databricks122

```
tests               | 2023-07-18 11:47:28 WARNING: not retrying; stacktrace does not match pattern '\Q[Databricks][DatabricksJDBCDriver](500593) Communication link failure. Failed to connect to server. Reason: HTTP retry after response received with no Retry-After header, error: HTTP Response code: 503, Error message: Unknown.': [java.lang.AssertionError: Expected callback to throw QueryExecutionException
tests               | 	at io.trino.tempto.assertions.QueryAssert.assertQueryFailure(QueryAssert.java:113)
tests               | 	at io.trino.tests.product.deltalake.TestDeltaLakeDatabricksDelete.testDeletionVectors(TestDeltaLakeDatabricksDelete.java:89)
tests               | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
tests               | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
tests               | 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
tests               | 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
tests               | 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
tests               | 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
tests               | 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
tests               | 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
tests               | 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
tests               | 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
tests               | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
tests               | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
tests               | 	at java.base/java.lang.Thread.run(Thread.java:833)
tests               | ]
tests               | 2023-07-18 11:47:28 INFO: Test io.trino.tests.product.deltalake.TestDeltaLakeDatabricksDelete.testDeletionVectors took 39.69s
tests               | 2023-07-18 11:47:28 INFO: FAILURE     /    io.trino.tests.product.deltalake.TestDeltaLakeDatabricksDelete.testDeletionVectors (Groups: profile_specific_tests, delta-lake-exclude-91, delta-lake-databricks, delta-lake-exclude-104, delta-lake-exclude-113, delta-lake-exclude-73) took 40.3 seconds
tests               | 2023-07-18 11:47:28 SEVERE: Failure cause:
tests               | java.lang.AssertionError: Expected callback to throw QueryExecutionException
tests               | 	at io.trino.tempto.assertions.QueryAssert.assertQueryFailure(QueryAssert.java:113)
tests               | 	at io.trino.tests.product.deltalake.TestDeltaLakeDatabricksDelete.testDeletionVectors(TestDeltaLakeDatabricksDelete.java:89)
tests               | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
tests               | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
tests               | 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
tests               | 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
tests               | 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
tests               | 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
tests               | 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
tests               | 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
tests               | 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
tests               | 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
tests               | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
tests               | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
tests               | 	at java.base/java.lang.Thread.run(Thread.java:833)
```

https://github.com/trinodb/trino/actions/runs/5583708174/jobs/10204496996?pr=18320

--- a/testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksDelete.java
+++ b/testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksDelete.java
@@ -82,11 +82,11 @@ public class TestDeltaLakeDatabricksDelete
 
             assertThat(onTrino().executeQuery("SHOW TABLES FROM delta.default"))
                     .contains(row(tableName));
+            assertThat(onTrino().executeQuery("SELECT version, operation FROM delta.default.\"" + tableName + "$history\""))
+                    .containsOnly(row(0, "CREATE TABLE"), row(1, "WRITE"), row(2, "DELETE"));
             assertThat(onTrino().executeQuery("SELECT comment FROM information_schema.columns WHERE table_schema = 'default' AND table_name = '" + tableName + "'"))
                     .hasNoRows();
             assertQueryFailure(() -> onTrino().executeQuery("SELECT * FROM delta.default." + tableName))
-                    .hasMessageMatching(".* Table .* does not exist");
-            assertQueryFailure(() -> onTrino().executeQuery("SELECT * FROM delta.default.\"" + tableName + "$history\""))
                     .hasMessageMatching(".* Table .* does not exist");
             assertQueryFailure(() -> onTrino().executeQuery("SHOW COLUMNS FROM delta.default." + tableName))
                     .hasMessageMatching(".* Table .* does not exist");

